### PR TITLE
ROC-7140 Add appinsights to LA journey

### PR DESCRIPTION
**Pull request by Valkrys**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-7148

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/appinsights/AppInsights.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4840061&bid=16071384)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/appinsights/AppInsights.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4840061&bid=16071384)

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java [hunk 1]
```diff
@@ -104,7 +106,15 @@ public void save(
...
-                updatedClaim, LocalDateTime.now(), authorization);
+                    updatedClaim, LocalDateTime.now(), authorization);
+            updatedClaim = claimService.getClaimByExternalId(externalId, authorization);
+        }
+
+        if (FeaturesUtils.isOnlineDQ(updatedClaim)) {
+            appInsights.trackEvent(AppInsightsEvent.BOTH_PARTIES_ONLINE_DQ, REFERENCE_NUMBER, updatedClaim.getReferenceNumber());
+            updatedClaim = claimService.getClaimByExternalId(externalId, authorization);
+        } else {
+            appInsights.trackEvent(AppInsightsEvent.BOTH_PARTIES_OFFLINE_DQ, REFERENCE_NUMBER, updatedClaim.getReferenceNumber());
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java]: Raising the app insights events should been done in the raiseAppInsightsEvents method. Also the calls to `getClaimByExternalId` aren't needed

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java [hunk 2]
```diff
@@ -160,27 +170,23 @@ private void formaliseResponseAcceptance(
         }
     }
 
-    private void raiseAppInsightEvents(Claim claim, Response response, ClaimantResponse claimantResponse) {
+    private void raiseAppInsightEvents(Claim claim, Response response, ClaimantResponse claimantResponse, Optional<CaseEvent> caseEvent) {
         if (claimantResponse instanceof ResponseAcceptation) {
             appInsights.trackEvent(CLAIMANT_RESPONSE_ACCEPTED, REFERENCE_NUMBER, claim.getReferenceNumber());
         } else if (claimantResponse instanceof ResponseRejection) {
             if (isPartAdmissionOrIsStatePaidOrIsFullDefence(response)) {
-                raiseAppInsightEventForLegalAdvisorPilot(claim);
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java]: Not sure why you removed this

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java [hunk 3]
```diff
@@ -160,27 +170,23 @@ private void formaliseResponseAcceptance(
...
+                } else {
+                    appInsights.trackEvent(AppInsightsEvent.NON_LA_PILOT_ELIGIBLE, REFERENCE_NUMBER, claim.getReferenceNumber());
+                }
             }
         } else {
             throw new IllegalStateException("Unknown response type");
         }
     }
 
-    private void raiseAppInsightEventForLegalAdvisorPilot(Claim claim) {
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java]: And this

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java (hunk 1 above)
Comment (Valkrys) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java]: I'll remove it

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java (hunk 3 above)
Comment (Valkrys) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java]: is the changing of this okay?

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/appinsights/AppInsights.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4840061&bid=16071384)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/appinsights/AppInsights.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4840061&bid=16071384)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/appinsights/AppInsights.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4840061&bid=16071384)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/appinsights/AppInsights.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4840061&bid=16071384)

**Pull request closed without merging**